internal/core/src/segcore/ConcurrentVector.h

```diff
@@ -229,6 +230,27 @@ class ConcurrentVectorImpl : public VectorBase {
         return chunks_[chunk_index].data();
     }
 
+    // Heavy and expensive, keep this not frequently.
+    // This function should only be called when small index will be created.
+    // An index chunk corresponds to many data chunks, so multiple data chunks will be assembled into an index chunk.
+    Chunk
+    get_chunks(const std::vector<ssize_t>& chunk_ids) const {
+        std::vector<ssize_t> prefix_sums(1, 0);
```

Comment:
define prefix_sums with correct size to avoid resizing:
std::vector<ssize_t> prefix_sums(chunk_ids.size() + 1, 0)